kernel/work.c

```diff
@@ -599,6 +602,55 @@ bool k_work_cancel_sync(struct k_work *work,
 	return pending;
 }
 
+#if defined(CONFIG_WORKQUEUE_WORK_TIMEOUT)
+static void workto_handler(struct _timeout *to)
+{
+	struct k_work_q *queue = CONTAINER_OF(to, struct k_work_q, workto);
+	k_spinlock_key_t key;
+	const char *name;
+	struct k_work *work;
+	k_work_handler_t handler;
+
+	key = k_spin_lock(&lock);
+
+	flag_set(&queue->flags, K_WORK_QUEUE_BLOCKED_BIT);
+
+	name = k_thread_name_get(&queue->thread);
+	work = queue->work;
+	handler = work->handler;
+
+	if (name != NULL) {
+		LOG_WRN("queue %s blocked by work %p with handler %p", name, work, handler);
+	} else {
+		LOG_WRN("queue %p blocked by work %p with handler %p", queue, work, handler);
```

Comment:
Seems like the k_oops() from the last PR might be a better choice, though maybe configurable.   A mere warning message isn't likely loud enough for something we almost all agree is a should-never-happen condition.

Alternatively make the callback be settable by the app and have the default blow up, but let apps do what they want?

Reply:
changed strategy a bit, since the timeout is run from a _timeout, k_oops() makes no sense since it is not run from the same thread as the work queue, so instead, abort the work queue and let the kernel handle it (essential thread would result in k_panic())